model/atmosphere/dycore/tests/dycore_stencil_tests/test_compute_horizontal_velocity_quantities_and_fluxes.py

```diff
@@ -0,0 +1,260 @@
+# ICON4Py - ICON inspired code in Python and GT4Py
+#
+# Copyright (c) 2022-2024, ETH Zurich and MeteoSwiss
+# All rights reserved.
+#
+# Please, refer to the LICENSE file in the root directory.
+# SPDX-License-Identifier: BSD-3-Clause
+from typing import Any
+
+import gt4py.next as gtx
+import numpy as np
+import pytest
+
+import icon4py.model.common.utils.data_allocation as data_alloc
+from icon4py.model.atmosphere.dycore.stencils.compute_horizontal_velocity_quantities import (
+    compute_horizontal_velocity_quantities_and_fluxes,
+)
+from icon4py.model.common import dimension as dims
+from icon4py.model.common.grid import base, horizontal as h_grid
+from icon4py.model.common.states import utils as state_utils
+from icon4py.model.testing import helpers as test_helpers
+
+from .test_compute_avg_vn_and_graddiv_vn_and_vt import (
+    compute_avg_vn_and_graddiv_vn_and_vt_numpy,
+)
+from .test_compute_contravariant_correction import (
+    compute_contravariant_correction_numpy,
+)
+from .test_compute_derived_horizontal_winds_and_ke_and_horizontal_advection_of_w_and_contravariant_correction import (
+    extrapolate_to_surface_numpy,
+)
+from .test_compute_horizontal_kinetic_energy import compute_horizontal_kinetic_energy_numpy
+from .test_compute_mass_flux import (
+    compute_mass_flux_numpy,
+)
+from .test_interpolate_vn_to_half_levels_and_compute_kinetic_energy_on_edges import (
+    interpolate_vn_to_half_levels_and_compute_kinetic_energy_on_edges_numpy,
+)
+from .test_interpolate_vt_to_interface_edges import interpolate_vt_to_interface_edges_numpy
+
+
+def compute_vt_vn_on_half_levels_and_kinetic_energy_numpy(
+    connectivities: dict[gtx.Dimension, np.ndarray],
+    vn: np.ndarray,
+    tangential_wind: np.ndarray,
+    vn_on_half_levels: np.ndarray,
+    tangential_wind_on_half_levels: np.ndarray,
+    horizontal_kinetic_energy_at_edges_on_model_levels: np.ndarray,
+    wgtfac_e: np.ndarray,
+    wgtfacq_e: np.ndarray,
+    nlevp1: int,
+) -> tuple[np.ndarray, ...]:
+    k = np.arange(nlevp1)
+    k = k[np.newaxis, :]
```

Comment:
I would merge these two lines but up to you:
```suggestion
    k = np.arange(nlevp1)[np.newaxis, :]
```